presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -165,16 +261,33 @@ public synchronized AllNodes getAllNodes()
     }
 
     @Override
-    public Set<Node> getActiveNodes()
+    public Set<Node> getNodes(NodeState state)
     {
-        return getAllNodes().getActiveNodes();
+        switch (state) {
+            case ACTIVE:
+                return getAllNodes().getActiveNodes();
+            case INACTIVE:
+                return getAllNodes().getInactiveNodes();
+            case SHUTTING_DOWN:
+                return getAllNodes().getShuttingDownNodes();
+            default:
+                throw new IllegalArgumentException("Unknown node state " + state);
+        }
     }
 
     @Override
-    public synchronized Set<Node> getActiveDatasourceNodes(String datasourceName)
+    public synchronized Set<Node> getNodes(NodeState state, String datasourceName)
     {
         refreshIfNecessary();
-        return activeNodesByDataSource.get(datasourceName);
+        switch (state) {
+            case ACTIVE:
+                return activeNodesByDataSource.get(datasourceName);
+            case INACTIVE:
```

Comment:
Does anyone call this function with `INACTIVE` or `SHUTTING_DOWN`? Since it returns a constant, those calls probably don't make sense. I would switch this back to `getActiveDatasourceNodes` unless there's a reason we need to support this